.github/workflows/ci-mingw.yml

```diff
@@ -0,0 +1,79 @@
+# This is a copy of ci-unix-shared-local.yml for building shared libraries
+# with an additional build configuration (using installed deps and dav1d).
+
+name: CI MinGW
+on: [push, pull_request]
+
+permissions:
+  contents: read
+
+# Cancel the workflow if a new one is triggered from the same PR, branch, or tag, except on main.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+jobs:
+  build-mingw:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        sys:
+          - mingw64
+          - ucrt64
+          - clang64
+
+    defaults:
+      run:
+        shell: msys2 {0}
+    steps:
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+    - uses: msys2/setup-msys2@v2
+      with:
+        msystem: ${{ matrix.sys }}
+        update: true
+        install: >-
+          git
+        pacboy: >-
+          aom:p
+          cmake:p
+          diffutils:p
+          cc:p
+          dav1d:p
+          imagemagick:p
+          libjpeg-turbo:p
+          libpng:p
+          libyuv:p
+          ninja:p
+          zlib:p
+    - name: Cache external dependencies
+      id: cache-ext
+      uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
+      with:
+        path: ext
+        key: ${{ matrix.sys }}-mingw-${{ hashFiles('ext/*.cmd') }}
+    - name: Print cmake version
+      run: cmake --version
+    - name: Build GoogleTest
```

Comment:
Optional: Would be good to build libsharpyuv as well. Can do this in a follow-up pull request.

Reply:
Done